ai_powered_qa/custom_plugins/playwright_plugin.py

```diff
@@ -216,21 +308,39 @@ def clean_html(html_content):
     - We replace <div> with <d> as they are very common.
     - The _clean_attributes function removes all `data-` attributes
     """
-    html_clean = _clean_attributes(html_content)
+    soup = BeautifulSoup(html_content, "html.parser")
+    _remove_not_visible(soup)
+    _remove_useless_tags(soup)
+    _unwrap_single_child(soup)
+    html_clean = str(soup)
+    html_clean = _clean_attributes(html_clean)
     html_clean = re.sub(r"<div[\s]*>[\s]*</div>", "", html_clean)
     html_clean = re.sub(r"<!--[\s\S]*?-->", "", html_clean)
-    html_clean = html_clean.replace("<div", "<d").replace("</div>", "</d>")
-    soup = BeautifulSoup(html_clean, "html.parser")
-    _unwrap_single_child(soup)
-    _remove_useless_tags(soup)
-    return str(soup)
+    # html_clean = html_clean.replace("<div", "<d").replace("</div>", "</d>")
+    return html_clean
```

Comment:
Reordered to have access to all elements in soup. As the logic get's more complex, we might want to do it all in soup.